android/src/main/java/com/dooboolab/RNIap/RNIapModule.java

```diff
@@ -466,22 +466,46 @@ public void onPurchasesUpdated(BillingResult billingResult, @Nullable List<Purch
       return;
     }
 
-    Purchase purchase = purchases.get(0);
-
-    WritableMap item = Arguments.createMap();
-    item.putString("productId", purchase.getSku());
-    item.putString("transactionId", purchase.getOrderId());
-    item.putString("transactionDate", String.valueOf(purchase.getPurchaseTime()));
-    item.putString("transactionReceipt", purchase.getOriginalJson());
-    item.putString("purchaseToken", purchase.getPurchaseToken());
-    item.putString("dataAndroid", purchase.getOriginalJson());
-    item.putString("signatureAndroid", purchase.getSignature());
-    item.putBoolean("autoRenewingAndroid", purchase.isAutoRenewing());
-    item.putBoolean("isAcknowledgedAndroid", purchase.isAcknowledged());
-    item.putInt("purchaseStateAndroid", purchase.getPurchaseState());
-
-    sendEvent(reactContext, "purchase-updated", item);
-    DoobooUtils.getInstance().resolvePromisesForKey(PROMISE_BUY_ITEM, item);
+    if (purchases != null) {
+      for (Purchase purchase : purchases) {
+
+        WritableMap item = Arguments.createMap();
+        item.putString("productId", purchase.getSku());
+        item.putString("transactionId", purchase.getOrderId());
+        item.putString("transactionDate", String.valueOf(purchase.getPurchaseTime()));
+        item.putString("transactionReceipt", purchase.getOriginalJson());
+        item.putString("purchaseToken", purchase.getPurchaseToken());
+        item.putString("dataAndroid", purchase.getOriginalJson());
+        item.putString("signatureAndroid", purchase.getSignature());
+        item.putBoolean("autoRenewingAndroid", purchase.isAutoRenewing());
+        item.putBoolean("isAcknowledgedAndroid", purchase.isAcknowledged());
+        item.putInt("purchaseStateAndroid", purchase.getPurchaseState());
+
+        sendEvent(reactContext, "purchase-updated", item);
+        DoobooUtils.getInstance().resolvePromisesForKey(PROMISE_BUY_ITEM, item);
+      }
+    }
+  }
+
+  private void sendUnconsumedPurchases(final Promise promise) {
+    ensureConnection(promise, new Runnable() {
+      @Override
+      public void run() {
+        Purchase.PurchasesResult purchasesResult = billingClient.queryPurchases(BillingClient.SkuType.INAPP);
```

Comment:
@solidfox Hey is there a reason that you are only fetching `unconsumedPurchases` for only `INAPP`? What about `SUBS`?

Reply:
Since I am only using INAPP in my implementation I had no way to test anything else to a satisfactory degree but you are correct that the same procedure should probably be applied for non-consumables (iOS), downloadables, and subscriptions.